services/app-api/serverless.yml

```diff
@@ -34,18 +34,16 @@ custom:
   tableArn: ${env:AMENDMENTS_TABLE_ARN, cf:database-${self:custom.stage}.AmendmentsTableArn}
   atomicCounterTableName: ${env:AMENDMENTS_COUNTER_TABLE_NAME, cf:database-${self:custom.stage}.AmendmentsAtomicCounterTableName}
   atomicCounterTableArn: ${env:AMENDMENTS_COUNTER_TABLE_ARN, cf:database-${self:custom.stage}.AmendmentsAtomicCounterTableArn}
-  warmupEnabled:
-    production: true
-    development: false
+
   warmup:
-    enabled: ${self:custom.warmupEnabled.${self:custom.infrastructureType}}
+    enabled: true
     role: LambdaWarmupRole
     vpc: false
     events:
-      - schedule: rate(4 minutes)
+      - schedule: ${ssm:/configuration/${self:custom.stage}/warmup/schedule~true, ssm:/configuration/default/warmup/schedule~true, rate(4 minutes)}
```

Comment:
I tested this and it doesn't appear to work as intended.  An environment I deployed with this code is not showing any lambda warmup invocations.  My test procedure might be flawed, but I stepped through it twice.

I think what's happening is rate(4 minutes) is getting evaluated as a plain text string and passed to the warmup plugin.  The deployment isn't failing, but the warmup doesn't seem to be happening.
I think to get around this, you wand to do  
`schedule: rate(${ssm..., ssm..., "4 minutes"})`
So in this way, rate() will get correctly evaluated... the ssm overrides may specify a plain text string (maybe like 1 minute) if they want... otherwise the default is a string "4 minutes" that gets correctly evaluated by the rate() function.

Reply:
I confirmed that this is working. I watched the Cloudwatch log that was created (/aws/lambda/app-api-t171-create).
I can see the following logs in CW within 10 mins:

Message
No older events at this moment. 
Retry
START RequestId: 1ca0f72a-d072-4edb-a79b-7cd6e4498b00 Version: $LATEST
2021-04-29T22:38:02.363Z	1ca0f72a-d072-4edb-a79b-7cd6e4498b00	INFO	Warmed up!
END RequestId: 1ca0f72a-d072-4edb-a79b-7cd6e4498b00
REPORT RequestId: 1ca0f72a-d072-4edb-a79b-7cd6e4498b00	Duration: 8.88 ms	Billed Duration: 9 ms	Memory Size: 1024 MB	Max Memory Used: 85 MB	Init Duration: 426.81 ms	
START RequestId: 3339df05-13c7-4ba9-ab1d-2ac40f157196 Version: $LATEST
2021-04-29T22:38:02.452Z	3339df05-13c7-4ba9-ab1d-2ac40f157196	INFO	Warmed up!
END RequestId: 3339df05-13c7-4ba9-ab1d-2ac40f157196
REPORT RequestId: 3339df05-13c7-4ba9-ab1d-2ac40f157196	Duration: 2.52 ms	Billed Duration: 3 ms	Memory Size: 1024 MB	Max Memory Used: 86 MB	
START RequestId: e80ae406-a7b4-479d-b5bf-3aad651e03b2 Version: $LATEST
2021-04-29T22:38:02.463Z	e80ae406-a7b4-479d-b5bf-3aad651e03b2	INFO	Warmed up!
END RequestId: e80ae406-a7b4-479d-b5bf-3aad651e03b2
REPORT RequestId: e80ae406-a7b4-479d-b5bf-3aad651e03b2	Duration: 1.40 ms	Billed Duration: 2 ms	Memory Size: 1024 MB	Max Memory Used: 86 MB